packages/valory/skills/decision_maker_abci/behaviours/decision_receive.py

```diff
@@ -350,8 +356,18 @@ def _calculate_new_liquidity(self, net_bet_amount: int, vote: int) -> LiquidityI
             new_selected,
         )
 
+    def _compute_scaled_liquidity_measure(
+        self, token_amounts: List[int], token_prices: List[float]
+    ):
+        """Function to compute the scaled liquidity measure from token amounts and prices"""
+        return sum(
+            amount * price for amount, price in zip(token_amounts, token_prices)
+        ) / (10**18)
```

Comment:
This should exist in the input data. Otherwise, we are making the trader's code more complex without a reason since this is not necessary in the normal flow.

Reply:
the input data has just the initial values for:
* token amounts
* token prices
and scaled liquidity measure is always computed from the previous two values. When simulating trades we need to update the new values for token amounts and token prices in the market after each trade, these changes are computed on the fly during the simulation. In production the trader gets all these new values directly from the latest markets information, in simulation we need to compute manually those updates, including the liquidity measure. Hope my explanation is clear.

Reply:
It's clear, please make the precision a constant.

Reply:
I meant the token's precision :sweat_smile:

Reply:
token amounts are integers with no decimals so I guess you mean the tokens prices. ok done